server/use_cases/add_workshifts.py

```diff
@@ -1,14 +1,63 @@
 """
 This module contains the use case for adding work shifts.
 """
-def workshift_add_use_case(repo, work_shift):
+from datetime import datetime
+from domains.work_shift import WorkShift
+
+def workshift_add_use_case(repo, new_shift, existing_shifts):
     """
-    The function adds a work shift into the chosen database.
+    The function adds a work shift into the chosen database
+    after checking for overlaps with existing shifts.
     """
-    repo.add(work_shift)
-def workshift_add_multiple_use_case(repo, work_shifts):
+    if shift_already_exists(new_shift, existing_shifts):
+        return {"success": False,
+                "message": "You are signed up for another shift at this time"}
+
+    repo.add(new_shift)
+    return {"success": True, "message": "Shift added successfully"}
+
+def workshift_add_multiple_use_case(repo, work_shifts, existing_shifts):
     """
     The function adds multiple work shifts into the chosen database.
+    It checks each shift for overlap and only adds non-overlapping shifts.
     """
+    responses = []
     for work_shift in work_shifts:
-        workshift_add_use_case(repo, work_shift)
+        shift_id = (work_shift.code if isinstance(work_shift, WorkShift)
+            else work_shift["code"])
```

Comment:
Why are you checking if work_shift is an instance of WorkShift here? This looks like a band-aid, that covers up some internal problem. In this case, the work_shift should always be a JSON object or always be a WorkShift (not sometimes JSON, sometimes WorkShift).